content/master/concepts/compositions.md

```diff
@@ -1308,47 +1308,69 @@ parameters in the Composition.
 * Composition function `type` is `container`.
 * Composition function names are unique.
 
-### Resource schema validation
-<!-- vale write-good.TooWordy = NO -->
-Optionally, Crossplane can also validate the schema of the resources defined
-inside a Composition. This verifies that the resource `apiVersion` and `kinds`
-are valid. 
-<!-- vale write-good.TooWordy = YES -->
+### Composition schema aware validation
 
-Enable "schema validation" with the
-`--enable-composition-webhook-schema-validation` flag on the Crossplane pod.
+On top of the preceding integrity checks, Crossplane performs schema aware
+validation of Compositions, checking that `patches`, `readinessChecks` and
+`connectionDetails` are valid according to the involved resources' schemas, for
+example checking that the source/destination field of a patch is a valid field
+according to the source/destination resource's schema and that types match, if
+defined, taking into account transforms.
+
+{{<hint "note" >}}
+Composition schema aware validation is a beta feature, so Crossplane
+enables it by default. You can disable it by setting the
+`--enable-composition-webhook-schema-validation=false` flag on the Crossplane
+pod.
 The [Crossplane Pods]({{<ref "./pods#edit-the-deployment">}}) page has
 more information on enabling Crossplane flags.
+{{< /hint >}}
+
+#### Schema aware validation modes
+
+Integrity validation errors always result in a rejection of the Composition.
+
+You can configure how Crossplane should handle both missing involved resources'
+schemas and actual schema aware validation errors.
 
 {{<hint "note" >}}
-<!-- vale write-good.TooWordy = NO -->
-Schema validation only checks the `apiVersion` and `kind` are valid. Schema
-validation doesn't validate the fields of a specific resource.
-<!-- vale write-good.TooWordy = YES -->
+In case of any involved resource schema missing, Crossplane skips schema aware
+validation altogether.
```

Comment:
This is based on the validation mode according to the table below, right?

Reply:
That's for all modes, schema aware validation is not performed unless we have all required schemas available. Depending on the mode this could result in a warning or error.